### PR TITLE
Swapping Split-Transform order

### PR DESCRIPTION
**Pull request by VIGS25**
This PR addresses the changes in ordering between split and transform for about half of the files in `dc.molnet`. The remaining ones will be added, once this half has been reviewed. 

In the `muv_datasets.py`, there's an additional `task` based split option available. For the fold datasets returned, there is an indecision on whether to do the transformation ourselves, or allow the user to do it, since the train dataset varies for every fold. Currently, I return an empty list for the transformers and expect the user to use the transformer themselves based on what is used for the rest of the code.

@rbharath, @peastman: Reviews and suggestions requested.

### CODE REVIEW

**Comment by VIGS25** [comment]
These set of changes should be ready to merge in. 

A few points and questions though:

1. Would you prefer having the remaining half of the files under `load_function` in this PR, or a new one after this is merged?

2. What do you think of the `muv_datasets.py` case, which has an additional `task` based split - Would you want transform the folds inside `load_muv` or allow the user to do so?

3. I also noticed that MolNet does not have a provision to do cross validation intrinsically (barring a couple of load functions). In the examples across the library which do use cross validation, the transformed dataset are merged into different folds. The problem here lies in the datasets being transformed prior to cross validation. The usual procedure would be to take the full dataset and for each fold, split it into training and validation sets, transform the training set and apply the same transformation to the validation set.

**Comment by rbharath** [comment]
1. Let's just do a new PR to keep things simple

2. I'm not sure about the difference between the two options. What would each entail?

3. Yes, I think we don't have cross validation capabilities on Molnet right now. I think the main reason for this was compute time; running even the standard benchmarks uses up a lot of compute.
